components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -101,7 +101,7 @@ export class TypeORMUserDBImpl implements UserDB {
             creationDate: new Date().toISOString(),
             identities: [],
             additionalData: {
-                ideSettings: { defaultIde: "code" },
+                ideSettings: {},
```

Comment:
By removing `defaultIde: "code"` from the user default preferences, we can actually detect when the user actually select their `defaultIde`.

A workspace start will fallback anyway on `code` if the user does not select any preferred IDE.

On another note, we could entirely remove `ideSettings` instead of keeping it as empty object. My preference is being explicit for code readability purposes and keep it, but I am happy to get rid of it too, if we want to keep it clean.

Reply:
> By removing defaultIde: "code" from the user default preferences when we insert the user into the DB, we can actually detect when the user actually select their defaultIde.

Knowing when a user makes an explicit choice will be useful 🙏